pennylane/optimize/adaptive.py

```diff
@@ -184,7 +184,7 @@ def step_and_cost(self, circuit, operator_pool, drain_pool=False, params_zero=Tr
             to the step, and the largest gradient
         """
         cost = circuit()
-        device = circuit.device
+        original_circ = copy.copy(circuit)
```

Comment:
I think we can skip copying `circuit` three times by just defining the `qnode` here and then just replacing its `func` attributes.